src/shared/logs/logs.py

```diff
@@ -204,11 +263,15 @@ def get_dict(
 
         raw_events = self.get_raw_events(suggestion_ids)
 
+        # Cancellation pass - remove events that cancel each other within a given time window (5 min by default)
+        filtered_events = self._remove_canceling_events(raw_events)
+
         grouped_activity_log: dict[
-            int, list[PackageChangeEvent | SuggestionChangeEvent]
+            int,
+            list[PackageChangeEvent | SuggestionChangeEvent | MaintainerChangeEvent],
```

Comment:
As often as we use the type, maybe let's aggregate this in a class (or simply a type that is made up of this union)?